From 57ce80821823cd2bf70b97132567f7a588642e92 Mon Sep 17 00:00:00 2001
From: Andreas Moog <amoog@ubuntu.com>
Date: Sun, 19 Jun 2011 22:03:10 +0200
Subject: [PATCH] LIBS go after objects to allow linking with ld --as-needed

---
 makefile |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/makefile b/makefile
index 0a9cc94..509c2ff 100755
--- a/makefile
+++ b/makefile
@@ -82,16 +82,16 @@ all: $(ALL)
 
 # linking the program.
 $(PROG): $(GAMEOBJS)
-	$(CXX) $(LIBS) $(GAMEOBJS) -o $(PROG)
+	$(CXX) $(GAMEOBJS) -o $(PROG) $(LIBS)
 	
 pak: $(PAKOBJS)
-	$(CXX) $(LIBS) $(PAKOBJS) -o pak
+	$(CXX) $(PAKOBJS) -o pak $(LIBS)
 
 %.mo: %.po
 	msgfmt -c -o $@ $<
 
 mapeditor: $(MAPOBJS)
-	$(CXX) $(LIBS) $(MAPOBJS) -o mapeditor
+	$(CXX) $(MAPOBJS) -o mapeditor $(LIBS)
 
 # cleaning everything that can be automatically recreated with "make".
 clean:
-- 
1.7.4.1

From: Guus Sliepen <guus@debian.org>
Date: Thu, 4 Aug 2011 14:08:47 +0000 (+0200)
Subject: Don't link pak tool with SDL.
X-Git-Url: http://blobwars.git.sourceforge.net/git/gitweb.cgi?p=blobwars%2Fblobwars;a=commitdiff_plain;h=bc656f813135bc127c2e2811b686198a316aae1d

Don't link pak tool with SDL.
---

diff --git a/makefile b/makefile
index 509c2ff..fbbc9a2 100755
--- a/makefile
+++ b/makefile
@@ -23,6 +23,7 @@ CXXFLAGS += -DPAKNAME=\"$(PAKNAME)\" -DPAKLOCATION=\"$(DATADIR)\" -DUNIX -DGAMEP
 CXXFLAGS += -DLOCALEDIR=\"$(LOCALEDIR)\" -DMEDAL_SERVER_HOST=\"$(MEDAL_SERVER_HOST)\" -DMEDAL_SERVER_PORT=$(MEDAL_SERVER_PORT)
 CXXFLAGS += $(CFLAGS) -Werror
 LIBS = `sdl-config --libs` -lSDL_mixer -lSDL_image -lSDL_ttf -lSDL_net -lz
+PAKLIBS = -lz
 
 OBJS += CAudio.o
 OBJS += CBoss.o
@@ -85,7 +86,7 @@ $(PROG): $(GAMEOBJS)
 	$(CXX) $(GAMEOBJS) -o $(PROG) $(LIBS)
 	
 pak: $(PAKOBJS)
-	$(CXX) $(PAKOBJS) -o pak $(LIBS)
+	$(CXX) $(PAKOBJS) -o pak $(PAKLIBS)
 
 %.mo: %.po
 	msgfmt -c -o $@ $<
-- 
1.7.4.1

From 3d312b6f5df0303dfadf1c6012b572e2ba958a19 Mon Sep 17 00:00:00 2001
From: Pacho Ramos <pacho@gentoo.org>
Date: Mon, 29 Aug 2011 20:36:16 +0200
Subject: [PATCH] Make build system respect LDFLAGS.

---
 makefile |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/makefile b/makefile
index fbbc9a2..eb37204 100755
--- a/makefile
+++ b/makefile
@@ -83,16 +83,16 @@ all: $(ALL)
 
 # linking the program.
 $(PROG): $(GAMEOBJS)
-	$(CXX) $(GAMEOBJS) -o $(PROG) $(LIBS)
+	$(CXX) $(LDFLAGS) $(GAMEOBJS) -o $(PROG) $(LIBS)
 	
 pak: $(PAKOBJS)
-	$(CXX) $(PAKOBJS) -o pak $(PAKLIBS)
+	$(CXX) $(LDFLAGS) $(PAKOBJS) -o pak $(PAKLIBS)
 
 %.mo: %.po
 	msgfmt -c -o $@ $<
 
 mapeditor: $(MAPOBJS)
-	$(CXX) $(MAPOBJS) -o mapeditor $(LIBS)
+	$(CXX) $(LDFLAGS) $(MAPOBJS) -o mapeditor $(LIBS)
 
 # cleaning everything that can be automatically recreated with "make".
 clean:
-- 
1.7.4.1

