diff -urN autoconvert-0.3.16.orig/autogb.c autoconvert-0.3.16/autogb.c
--- autoconvert-0.3.16.orig/autogb.c	2003-12-12 21:07:01.000000000 +0800
+++ autoconvert-0.3.16/autogb.c	2009-06-09 00:00:23.816080800 +0800
@@ -27,7 +27,7 @@
 
 #include <stdio.h>
 #include <getopt.h>
-#include "hz.h"
+#include "include/hz.h"
 
 
 /* Print help and exit */
diff -urN autoconvert-0.3.16.orig/hzconvert/b2g.c autoconvert-0.3.16/hzconvert/b2g.c
--- autoconvert-0.3.16.orig/hzconvert/b2g.c	2001-04-28 09:45:46.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/b2g.c	2009-06-09 00:01:24.407331191 +0800
@@ -23,7 +23,7 @@
  * 1999.10
  */ 
 //$Header: /home/prog/autoconvert/hzconvert/b2g.c,v 1.2 2001/04/28 01:45:46 ygh Exp $
-#include "hz.h"
+#include "../include/hz.h"
 
 #define	BtoG_bad1 0xa1
 #define	BtoG_bad2 0xf5
diff -urN autoconvert-0.3.16.orig/hzconvert/b2u.c autoconvert-0.3.16/hzconvert/b2u.c
--- autoconvert-0.3.16.orig/hzconvert/b2u.c	2004-03-22 01:28:57.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/b2u.c	2009-06-09 00:01:24.453330617 +0800
@@ -23,7 +23,7 @@
  *1999.10
  */       
 //$Header: /home/prog/autoconvert/hzconvert/b2u.c,v 1.6 2001/04/28 01:45:46 ygh Exp $
-#include "hz.h"
+#include "../include/hz.h"
 
 #define	BtoU_bad1 0xa1
 #define	BtoU_bad2 0xf5
diff -urN autoconvert-0.3.16.orig/hzconvert/b2u.c.orig autoconvert-0.3.16/hzconvert/b2u.c.orig
--- autoconvert-0.3.16.orig/hzconvert/b2u.c.orig	2001-04-28 09:45:46.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/b2u.c.orig	2009-06-09 00:01:24.456330650 +0800
@@ -23,7 +23,7 @@
  *1999.10
  */       
 //$Header: /home/prog/autoconvert/hzconvert/b2u.c,v 1.6 2001/04/28 01:45:46 ygh Exp $
-#include "hz.h"
+#include "../include/hz.h"
 
 #define	BtoU_bad1 0xa1
 #define	BtoU_bad2 0xf5
diff -urN autoconvert-0.3.16.orig/hzconvert/b5percent.c autoconvert-0.3.16/hzconvert/b5percent.c
--- autoconvert-0.3.16.orig/hzconvert/b5percent.c	2000-08-31 11:13:50.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/b5percent.c	2009-06-09 00:01:24.613330716 +0800
@@ -5,7 +5,7 @@
  The percentage is taken over all BIG5 code space.
 ****************************************************/
 #include <string.h>
-#include "zhstatis.h"
+#include "../include/zhstatis.h"
 
 /* maximum key range = 986, duplicates = 4 */
 
diff -urN autoconvert-0.3.16.orig/hzconvert/g2u.c autoconvert-0.3.16/hzconvert/g2u.c
--- autoconvert-0.3.16.orig/hzconvert/g2u.c	2001-04-28 09:45:46.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/g2u.c	2009-06-09 00:01:24.616330958 +0800
@@ -24,7 +24,7 @@
  */       
 //$Header: /home/prog/autoconvert/hzconvert/g2u.c,v 1.4 2001/04/28 01:45:46 ygh Exp $
 
-#include "hz.h"
+#include "../include/hz.h"
 
 #define	GtoU_bad1 0xa1
 #define	GtoU_bad2 0xbc
diff -urN autoconvert-0.3.16.orig/hzconvert/gbpercent.c autoconvert-0.3.16/hzconvert/gbpercent.c
--- autoconvert-0.3.16.orig/hzconvert/gbpercent.c	2000-08-31 11:13:50.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/gbpercent.c	2009-06-09 00:01:24.763330521 +0800
@@ -4,7 +4,7 @@
  400 Most frequently used GB chars. 
  ****************************************************/
 #include <string.h>
-#include "zhstatis.h"
+#include "../include/zhstatis.h"
 
 /* maximum key range = 991, duplicates = 5 */
 
diff -urN autoconvert-0.3.16.orig/hzconvert/hz2gb.c autoconvert-0.3.16/hzconvert/hz2gb.c
--- autoconvert-0.3.16.orig/hzconvert/hz2gb.c	2001-04-28 09:45:46.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/hz2gb.c	2009-06-09 00:01:24.765330985 +0800
@@ -24,7 +24,7 @@
 */
 //$Header: /home/prog/autoconvert/hzconvert/hz2gb.c,v 1.2 2001/04/28 01:45:46 ygh Exp $
 
-#include "hz.h"
+#include "../include/hz.h"
 
 #define	hzb(c)	((((c) & 0x7f) >= 0x21) && (((c) & 0x7f) <= 0x7e))
 
diff -urN autoconvert-0.3.16.orig/hzconvert/hztogb.c autoconvert-0.3.16/hzconvert/hztogb.c
--- autoconvert-0.3.16.orig/hzconvert/hztogb.c	2000-04-19 13:43:32.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/hztogb.c	2009-06-09 00:01:24.766331112 +0800
@@ -30,7 +30,7 @@
 //$Header: /home/prog/autoconvert/hzconvert/hztogb.c,v 1.1.1.1 2000/04/19 05:43:32 ygh Exp $
 /*
 #include <stdio.h>
-#include "hz.h"
+#include "../include/hz.h"
 
 char *hz2gb(char*s,int *plen,int inst,char *out)
 {
diff -urN autoconvert-0.3.16.orig/hzconvert/io.c autoconvert-0.3.16/hzconvert/io.c
--- autoconvert-0.3.16.orig/hzconvert/io.c	2000-04-19 13:43:32.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/io.c	2009-06-09 00:01:24.768331645 +0800
@@ -23,7 +23,7 @@
  *1999.10
  */       
 //$Header: /home/prog/autoconvert/hzconvert/io.c,v 1.1.1.1 2000/04/19 05:43:32 ygh Exp $
-#include "hz.h"
+#include "../include/hz.h"
 
 char *hzconvert (s, plen, psaved, dbcvrt)
      char *s;
diff -urN autoconvert-0.3.16.orig/hzconvert/judge3.c autoconvert-0.3.16/hzconvert/judge3.c
--- autoconvert-0.3.16.orig/hzconvert/judge3.c	2000-11-29 01:31:19.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/judge3.c	2009-06-09 00:01:24.770330992 +0800
@@ -30,8 +30,8 @@
 
 
 #include <stdio.h>
-#include "hz.h"
-#include "zhstatis.h"
+#include "../include/hz.h"
+#include "../include/zhstatis.h"
 
 
 int j_code3(char * buff, int count)
diff -urN autoconvert-0.3.16.orig/hzconvert/judge.c autoconvert-0.3.16/hzconvert/judge.c
--- autoconvert-0.3.16.orig/hzconvert/judge.c	2000-08-31 11:11:08.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/judge.c	2009-06-09 00:01:24.771331259 +0800
@@ -27,7 +27,7 @@
 //$Header: /home/prog/autoconvert/hzconvert/judge.c,v 1.2 2000/08/31 03:11:08 ygh Exp $
 #include <stdio.h>
 
-#include "hz.h"
+#include "../include/hz.h"
 
 int j_code(char * buff,int count)
 {
diff -urN autoconvert-0.3.16.orig/hzconvert/search.c autoconvert-0.3.16/hzconvert/search.c
--- autoconvert-0.3.16.orig/hzconvert/search.c	2001-04-28 09:45:46.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/search.c	2009-06-09 00:01:24.775330790 +0800
@@ -19,7 +19,7 @@
 */
 #include <stdio.h>
 #include <string.h>
-#include "hz.h"
+#include "../include/hz.h"
 
 #define MAX 100
 
diff -urN autoconvert-0.3.16.orig/hzconvert/unicode.c autoconvert-0.3.16/hzconvert/unicode.c
--- autoconvert-0.3.16.orig/hzconvert/unicode.c	2001-04-28 09:45:46.000000000 +0800
+++ autoconvert-0.3.16/hzconvert/unicode.c	2009-06-09 00:01:24.780331426 +0800
@@ -24,7 +24,7 @@
  */       
 //$Header: /home/prog/autoconvert/hzconvert/unicode.c,v 1.5 2001/04/28 01:45:46 ygh Exp $
 
-#include "hz.h"
+#include "../include/hz.h"
 
 /*** UTF8 => UNICODE ***/
 
