--- oxim-1.4.1/configure.ac~	2009-11-15 00:15:02.166255238 +0800
+++ oxim-1.4.1/configure.ac	2009-11-15 00:29:14.850266704 +0800
@@ -253,6 +253,7 @@
 #-----------------------------------------------------------------------
 #  Check for Gtk2 Library ***
 #-----------------------------------------------------------------------
+gtk_binary_version=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
 AC_ARG_ENABLE([gtk-immodule],
     AS_HELP_STRING(
         [--enable-gtk-immodule],
@@ -260,13 +261,19 @@
         [],
         [enable_gtk_immodule="yes"])
 
+AC_ARG_WITH([gtk-immodule-dir],
+    AS_HELP_STRING(
+    	[--with-gkt-immodule-dir=PATH],
+	[set the gtk immodule directory]),
+	[test x$withval != x && gtk_immdoule_dir="$withval"],
+	[gtk_immodule_dir="$libdir"/gtk-2.0/$gtk_binary_version/immodules])
 if test "$enable_gtk_immodule" = "yes" ; then
     PKG_CHECK_MODULES(GTK2, gtk+-2.0 >= 2.2.0, use_gtk2="yes", use_gtk2="no")
     if test "$use_gtk2" = "no" ; then
 	AC_MSG_WARN([Gtk2 version must be > 2.2.0!])
 	enable_gtk_immodule="no"
     fi
-    GTK_IM_MODULE_DIR=`$PKG_CONFIG gtk+-2.0 --variable=libdir`/gtk-2.0/immodules
+    GTK_IM_MODULE_DIR=$gtk_immodule_dir
     AC_SUBST(GTK_IM_MODULE_DIR)
 fi
 AM_CONDITIONAL(ENABLE_GTK_IMMODULE, [ test "$enable_gtk_immodule" = "yes" ])
