--- fcitx-scm/doc/Makefile.am.origin	2009-08-17 17:41:02.609491677 +0800
+++ fcitx-scm/doc/Makefile.am	2009-08-17 17:43:41.080243234 +0800
@@ -1,5 +1,4 @@
-docdir = $(pkgdatadir)/doc
-doc_DATA = \
-        cjkvinput.txt wb_fh.htm fcitx3.pdf pinyin.txt fcitx_trunk.doxyfile \
+dist_doc_DATA = \
+        cjkvinput.txt wb_fh.htm fcitx3.pdf pinyin.txt \
 	fcitx3.odt Develop_Readme.GBK API.txt
-EXTRA_DIST = $(doc_DATA)
+EXTRA_DIST = fcitx_trunk.doxyfile
