From 78676351ee9a54ed8ccd2ce3550b593f27bc051d Mon Sep 17 00:00:00 2001
From: "Wulf C. Krueger" <philantrop@exherbo.org>
Date: Sat, 13 Apr 2013 19:13:39 +0200
Subject: [PATCH] Fix SDL check

---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index ae5fa35..786a913 100644
--- a/configure.in
+++ b/configure.in
@@ -133,7 +133,7 @@ fi
 
 dnl Check for SDL
 SDL_VERSION=1.2.0
-AM_PATH_SDL2($SDL_VERSION,
+AM_PATH_SDL($SDL_VERSION,
             :,
 	    AC_MSG_ERROR([*** SDL version $SDL_VERSION not found!])
 )
-- 
1.8.2.1

