--- scim-hangul-0.3.2/src/scim_hangul_imengine.cpp.orig	2009-02-10 21:53:54.000000000 +0800
+++ scim-hangul-0.3.2/src/scim_hangul_imengine.cpp	2009-02-10 21:51:53.000000000 +0800
@@ -33,6 +33,7 @@
   #include <config.h>
 #endif
 
+#include <string.h>
 #include <scim.h>
 #include "scim_hangul_imengine.h"
 
--- scim-hangul-0.3.2/src/scim_hangul_imengine_setup.cpp.orig	2009-02-10 22:02:33.000000000 +0800
+++ scim-hangul-0.3.2/src/scim_hangul_imengine_setup.cpp	2009-02-10 22:02:51.000000000 +0800
@@ -34,6 +34,7 @@
   #include <config.h>
 #endif
 
+#include <string.h>
 #include <scim.h>
 #include <gtk/scimkeyselection.h>
 
